bootstrap/kubeadm/api/v1beta1/kubeadmconfiglist_webhook.go

```diff
@@ -20,6 +20,8 @@ import (
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+// SetupWebhookWithManager sets up KubeadmConfigList webhooks.
+// Deprecated: This method is going to be removed in a next release.
```

Comment:
Q: if I remember well we recently implemented a check to ensure all the webhooks exist; Have checked if this deprecation conflates with this check?

Also given that this is golang code and not API types (CRD), can we queue up this for 1.1?

Reply:
> Q: if I remember well we recently implemented a check to ensure all the webhooks exist; Have checked if this deprecation conflates with this check?

The check verifies that the types implement the conversion interface correct (aka they have ConvertTo/ConvertFrom/Hub). So they will still pass after this func is dropped, in fact for all other `*List` types these functions don't exist right now.

> Also given that this is golang code and not API types (CRD), can we queue up this for 1.1?

I think based on our policies it should be okay to deprecate those funcs with v1.1.0 and drop them with v1.2.0. If I got it correctly we have to deprecate with a `.0` release (because deprecations are not cherry-pickable) so we can drop them with the next following minor release.

I'm not sure I understood the question correctly.

Reply:
I'd remove these now given that these are mostly for internal controller management and they don't do much

Reply:
Fine for me. Everyone who might use them can essentially just stop doing it with v1.1 without any downsides.

Reply:
Updated the PR accordingly